src/config/configAggregator.ts

```diff
@@ -116,16 +116,27 @@ export class ConfigAggregator extends AsyncOptionalCreatable<ConfigAggregator.Op
 
   // Use typing from AsyncOptionalCreatable to support extending ConfigAggregator.
   // We really don't want ConfigAggregator extended but typescript doesn't support a final.
+  public static async create<T extends ConfigAggregator>(
```

Comment:
this makes the types cleaner inside the method, but is kinda ugly
explanation: because this could be subclassed, the original let anything that extends AsyncOptionalCreatable use it.

we can have a tighter condition , where if we know the thing is really a ConfigAggregator (or its subclasses) the types can be more specific.

I'd also be happy to let `create` only work with ConfigAggregator but I don't understand heavy-OOP to know the implications.  There are a few `extends ConfigAggregator` I saw in the wild on github so I wanted to be careful with the public API.